packages/trpc/server/routers/viewer/eventTypes/getEventTypeOptions.handler.ts

```diff
@@ -0,0 +1,245 @@
+import { checkRateLimitAndThrowError } from "@calcom/lib/checkRateLimitAndThrowError";
+import { getUserAvatarUrl } from "@calcom/lib/getAvatarUrl";
+import { getBookerBaseUrlSync } from "@calcom/lib/getBookerUrl/client";
+import { getBookerBaseUrl } from "@calcom/lib/getBookerUrl/server";
+import { EventTypeRepository } from "@calcom/lib/server/repository/eventType";
+import { MembershipRepository } from "@calcom/lib/server/repository/membership";
+import { ProfileRepository } from "@calcom/lib/server/repository/profile";
+import type { PrismaClient } from "@calcom/prisma";
+import { MembershipRole, SchedulingType } from "@calcom/prisma/enums";
+import { teamMetadataSchema } from "@calcom/prisma/zod-utils";
+import { EventTypeMetaDataSchema } from "@calcom/prisma/zod-utils";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TrpcSessionUser } from "../../../trpc";
+import { listOtherTeamHandler } from "../organizations/listOtherTeams.handler";
+import type { TGetEventTypeOptionsSchema } from "./getEventTypeOptions.schema";
+
+type GetEventTypeOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+    prisma: PrismaClient;
+  };
+  input: TGetEventTypeOptionsSchema;
+};
+
+type Option = {
+  value: string;
+  label: string;
+};
+
+type res = Awaited<
+  ReturnType<typeof MembershipRepository.findAllByUpIdIncludeMinimalEventTypes>
+>[number]["team"]["eventTypes"][number];
+
+type EventType = Omit<res, "forwardParamsSuccessRedirect"> & { children?: { id: number }[] };
+
+export const getEventTypeOptions = async ({ ctx, input }: GetEventTypeOptions) => {
+  await checkRateLimitAndThrowError({
+    identifier: `eventTypes:getEventTypeOptions:${ctx.user.id}`,
+    rateLimitingType: "common",
+  });
+
+  const user = ctx.user;
+  const teamId = input?.teamId;
+  const isOrg = input?.isOrg;
+
+  const skipTeamOptions = !isOrg;
+  const skipEventTypes = !!isOrg;
+
+  const userProfile = ctx.user.profile;
+  const profile = await ProfileRepository.findByUpId(userProfile.upId);
+  const parentOrgHasLockedEventTypes =
+    profile?.organization?.organizationSettings?.lockEventTypeCreationForUsers;
+
+  const [profileMemberships, profileEventTypes] = await Promise.all([
+    MembershipRepository.findAllByUpIdIncludeMinimalEventTypes(
+      {
+        upId: userProfile.upId,
+      },
+      {
+        where: {
+          accepted: true,
+        },
+        skipEventTypes,
+      }
+    ),
+    teamId
+      ? []
+      : EventTypeRepository.findAllByUpIdWithMinmalData(
+          {
+            upId: userProfile.upId,
+            userId: user.id,
+          },
+          {
+            where: {
+              teamId: null,
+            },
+            orderBy: [
+              {
+                position: "desc",
+              },
+              {
+                id: "asc",
+              },
+            ],
+          }
+        ),
+  ]);
```

Comment:
this fetches user personal event types so we can skip it if teamId is passed as we only want team event types.